flepimop/gempyor_pkg/tests/seir/test_seir.py

```diff
@@ -16,8 +18,77 @@
 os.chdir(os.path.dirname(__file__))
 
 
+def test_check_parameter_positivity():
+
+    parameter_names = [
+        "alpha*1*1*1",
+        "sigma_OMICRON*1*1*1",
+        "3*gamma*1*1*1",
+        "epsilon+omegaph4*1*1*1",
+        "1*zeta*1*1",
+        "r0*gamma*theta10*1*chi_OMICRON*1",
+        "r0*gamma*theta9*1*chi_OMICRON*1",
+        "eta_X0toX3_highIE*1*1*nuage0to17",
+        "eta_X0toX3_highIE*1*1*nuage18to64LR",
+        "eta_X0toX3_highIE*1*1*nuage18to64HR",
+        "eta_X0toX3_highIE*1*1*nuage65to100",
+    ]
+    dates = pd.date_range("2023-03-19", "2025-04-30", freq="D")
+    subpop_names = [
+        "56000",
+        "50000",
+        "11000",
+        "02000",
+        "38000",
+        "46000",
+        "10000",
+        "30000",
+        "44000",
+        "23000",
+    ]
+
+    # No negative params
+    test_array1 = np.zeros(
+        (len(parameter_names) - 1, len(dates) - 1, len(subpop_names) - 1)
+    )
+
+    # Randomized negative params
+    test_array2 = np.zeros(
+        (len(parameter_names) - 1, len(dates) - 1, len(subpop_names) - 1)
+    )
+    for _ in range(5):
+        test_array2[randint(0, len(parameter_names) - 1)][randint(0, len(dates) - 1)][
+            randint(0, len(subpop_names) - 1)
+        ] = -1
+
+    # Set negative params with intentional redundancy
+    test_array3 = np.zeros((len(parameter_names), len(dates), len(subpop_names)))
+    randint_first_dim = randint(0, len(parameter_names) - 1)
+    randint_second_dim = randint(0, len(dates) - 2)
+    randint_third_dim = randint(0, len(subpop_names) - 1)
+    test_array3[0][0][0] = -1
+    test_array3[
+        randint(0, len(parameter_names) - 1),
+        randint(0, len(dates) - 1) :,
+        randint(0, len(subpop_names) - 1),
+    ] = -1
+    test_array3[randint_first_dim][randint_second_dim][randint_third_dim] = -1
+    test_array3[randint_first_dim][randint_second_dim + 1][randint_third_dim] = -1
+
+    seir.check_parameter_positivity(
+        test_array1, parameter_names, dates, subpop_names
+    )  # NoError
+
+    with pytest.raises(ValueError):
+        assert seir.check_parameter_positivity(
+            test_array2, parameter_names, dates, subpop_names
+        )  # ValueError
+        assert seir.check_parameter_positivity(
+            test_array3, parameter_names, dates, subpop_names
+        )  # ValueError
```

Comment:
1. Minor: Stylistic, but maybe we should move these assertions to their respective construction? So the code above becomes:
```python
    # No negative params
    test_array1 = np.zeros(
        (len(parameter_names) - 1, len(dates) - 1, len(subpop_names) - 1)
    )
    assert seir.check_parameter_positivity(test_array1, parameter_names, dates, subpop_names)  is None
```
2. This needs to be two separate calls to `pytest.raises`, I'm pretty sure as is the call to `test_array3` doesn't happen. Also we can drop the assert, pytest can introspect with the context manager.
3. Can we add error message matching to this? Or at least that we get the correct parameter name in the error message? I.e. the code is not producing false positives.